pkg/webhooks/webhookpatch.go

```diff
@@ -58,32 +57,25 @@ func patchMutatingWebhookConfig(client admissionregistrationv1beta1client.Mutati
 
 const delayedRetryTime = time.Second
 
-// Moved out of injector main. Changes:
-// - pass the existing k8s client
-// - use the K8S root instead of citadel root CA
-// - removed the watcher - the k8s CA is already mounted at startup, no more delay waiting for it
-func PatchCertLoop(injectionWebhookConfigName, webhookName, caBundlePath string, client kubernetes.Interface, stopCh <-chan struct{}) {
+// PatchCertLoop monitors webhooks labeled with the specified revision and patches them with the given CA bundle
+func PatchCertLoop(revision, injectionWebhookConfigName, webhookName, caBundlePath string, client kubernetes.Interface, stopCh <-chan struct{}) {
 	// K8S own CA
 	caCertPem, err := ioutil.ReadFile(caBundlePath)
 	if err != nil {
 		log.Errorf("Skipping webhook patch, missing CA path %v", caBundlePath)
 		return
 	}
 
-	var retry bool
-	if err = patchMutatingWebhookConfig(client.AdmissionregistrationV1beta1().MutatingWebhookConfigurations(),
-		injectionWebhookConfigName, webhookName, caCertPem); err != nil {
-		log.Warn("Error patching Webhook ", err)
-		retry = true
-	}
-
-	shouldPatch := make(chan struct{})
+	// used for retries on patching canonical injection webhook for this revision
+	shouldPatchCanonicalWebhook := make(chan struct{})
 
-	watchlist := cache.NewListWatchFromClient(
+	watchlist := cache.NewFilteredListWatchFromClient(
 		client.AdmissionregistrationV1beta1().RESTClient(),
 		"mutatingwebhookconfigurations",
 		"",
-		fields.ParseSelectorOrDie(fmt.Sprintf("metadata.name=%s", injectionWebhookConfigName)))
+		func(options *metav1.ListOptions) {
+			options.LabelSelector = fmt.Sprintf("istio.io/rev=%s", revision)
```

Comment:
major nit: I  think there is a `label.Revision` constant defined in istio/api

Reply:
Changed, thanks